File: leetcode-submissions/2661_First_Completely_Painted_Row_or_Column.cpp
class Solution {
public:
    int firstCompleteIndex(vector<int>& arr, vector<vector<int>>& mat) {
        // we need to find the min indexes of to paint the columns 
        unordered_map<int,int>mp;
        int m=mat.size() ,n=mat[0].size();
        for(int i=0;i<arr.size();i++){
            mp[arr[i]]=i;
        }
        //we stored the indexs of the arrays elements in the map 
        int minIndex=INT_MAX;
        for(int row=0;row<m;row++){
            int lastIndex=INT_MIN;
            
            for(int col=0;col<n;col++){
                int index=mp[mat[row][col]];
                lastIndex=max(lastIndex,index);
            }
            minIndex=min(minIndex,lastIndex);
        }
        for(int col=0;col<n;col++){
            int lastIndex=INT_MIN;
            for(int row=0;row<m;row++){
                int index=mp[mat[row][col]];
                lastIndex=max(lastIndex,index);
            }
            minIndex=min(minIndex,lastIndex);
        }
        return minIndex;

    }
};

File: leetcode-submissions/2661_First_Completely_Painted_Row_or_Column.ts
function firstCompleteIndex(arr: number[], mat: number[][]): number {
    const map:Map<number,[number,number]>=new Map();
    for(let i:number=0;i<mat.length;i++){
        for(let j:number =0;j<mat[0].length;j++){
            map.set(mat[i][j],[i,j]);
        }
    }
    for(let i:number =0;i<arr.length;i++){
        const [row,col]=map.get(arr[i]);
        mat[row][col]*=-1;
        if(checkRowPainted(row,mat) || checkColPainted(col,mat)){
            return i;
        }
    }
    return -1;
};
function checkRowPainted(row:number,mat:number[][]):boolean{
    for(let col:number=0;col<mat[0].length;col++){
        if(mat[row][col]>0){
            return false;
        }
    }
    return true ;
}
function checkColPainted(col:number ,mat:number[][]):boolean{
    for(let row:number =0;row<mat.length;row++){
         if(mat[row][col]>0){
            return false;
        }
    }
    return true ;
}